File: backend/components/metiers/threed/tests_centralises/api/samples/core/test_sample_core.js
// test_sample_core.js – Test ultra avancé des samples cœur métier API Threed (centralisé)

const { sampleCoreLogic } = require('../../../api/samples/core');

describe('Samples Cœur métier API Threed', () => {
  it('doit exécuter la logique cœur métier ultra avancée', () => {
    const data = { input: 'test', flag: true };
    const result = sampleCoreLogic(data);
    expect(result.core_processed).toBe(true);
    expect(result.input).toEqual(data);
    expect(result.status).toBe('ok');
  });
});

File: backend/components/metiers/threed/tests_centralises/api/samples/core/test_sample_core.py
import unittest
from backend.components.metiers.threed.api.samples.core import sample_core_logic


class TestSampleCore(unittest.TestCase):
    def test_core_logic(self):
        # TODO: Test de la logique métier core
        self.assertTrue(True)

    def test_core_integration(self):
        # TODO: Test d'intégration core
        self.assertEqual(42, 42)

    def test_import_sample_core_logic(self):
        res = sample_core_logic({"x": 1})
        self.assertIn("core_processed", res)
